src/lib.rs

```diff
@@ -509,6 +511,107 @@ impl<'a, T: CBufItem, const SIZE: usize> CBufReader<'a, T, SIZE> {
         self.next_local
     }
 
+    /// Sets the index used for sequential reads.
+    ///
+    /// (i.e., [`fetch_next_item`](Self::fetch_next_item)
+    /// and [`available_items_iter`](Self::available_items_iter)).
+    #[inline]
+    pub fn set_next_index(&mut self, next: CBufIndex<SIZE>) {
+        self.next_local = next;
+    }
+
+    /// Finds the earliest valid index that makes the predicate true
+    /// assuming buffer contents are sorted by `predicate()` in `false`, `true` order.
+    ///
+    /// If `fromstart` is set, searches all available data, otherwise
+    /// the search starts at the `current_next_index()`.
+    ///
+    /// On completion, the `current_next_index()` is set to
+    /// the returned index, or to the end index if
+    /// no true value is found and InvalidIndexError.
+    /// if buffer contents are sorted by `predicate()` in `false`, `true` order.
```

Comment:
I previously mentioned to you (in a different venue) that I wasn't sure `search` should mutate the `CBufReader`. Looking at this again, I feel that it probably should _not_.

Some explanation is probably in order. I see `CBufReader` as being something you use to read sequentially from the `CBuf`, and this sort of violates the spirit of that by causing the same `CBufReader` to possibly jump back and forth in the buffer.

That said, that leaves the question of what `search` _should_ return instead. How about this idea: on success, it returns a `CBufReader` with `next_local` set to the index `search` currently sets in `self`. Seem reasonable?

Reply:
I agree that search should not mutate the `CBufReader`.  Reasonable things it could return on success are:
1. A `CBufIndex`
2. A `CBufReader` copied from `self` and then set with `next_local` at the location.  (Your suggestion.)

I think that I agree that 2. is probably better.

As a nicety, is it possible (good?) for the lifetime of the returned `CBufReader` to not be the lifetime of `self`, but rather be the lifetime of the `CBuf` that `self` got its lifetime from?

Reply:
@dmopalmer has gone and implemented option (1). That's fine by me.